### PR TITLE
Add 2023 solutions

### PR DESCRIPTION
**Pull request by justapotato213**
edited docs, and added code

### CODE REVIEW

File: 2023/s1.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    tape -= 2
+
+# Checking for top-bottom adjacency
+for i in range(c):
+  # If the triangles are pointing into each other, they don't share an edge
+  # Referring to the diagram, this only happens on even triangle columns (starting at 0)
+  if row1[i] and row2[i] and i % 2 == 0:
+    tape -= 2
+
+print(tape)
```
Comment (AritroSaha10) [inline 2023/s1.py:26]: Can you add a newline as the last line?

**Comment by justapotato213** [comment]
Everything *should* be there and formatted properly, lmk if there's anything missing.

File: 2023/s2.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# solve
+ans = [float('infinity')] * N
+ans[0] = 0
+for j in range(1, N):
+    for i in range(j - 1, -1, -1):
+        grid[i][j] = grid[i + 1][j - 1] + abs(hills[i] - hills[j])
+        if grid[i][j] < ans[j - i]:
+            ans[j - i] = grid[i][j]
+
+print(" ".join(str(i) for i in ans))
```
Comment (AritroSaha10) [inline 2023/s2.py:19]: newline as last line

File: 2023/s3.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            grid[i][-1] = "e"
+
+    elif C + count == M and R != 0:
+        for i in range((N - R) // 2):
+            grid[i][0] = "e"
+            grid[N - i - 1][0] = 'e'
+
+    # print ans
+    for row in grid:
+        print("".join(str(i) for i in row))
```
Comment (AritroSaha10) [inline 2023/s3.py:75]: newline as last line

File: 2023/s3.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    print("IMPOSSIBLE")
+else:
+    # col palindromes
+    count = 0
+    if C % 2 == 1 and M % 2 == 1:
+        for i in range(N):
+            grid[i][M//2] = 'a'
+        C -= 1
+        count += 1
+
```
Comment (AritroSaha10) [inline 2023/s3.py:23]: remove newline

**Review by AritroSaha10 — CHANGES_REQUESTED** [review]

**Comment by justapotato213** [comment]
@AritroSaha10 fixed

**Review by AritroSaha10 — APPROVED** [review]

**Pull request merged**